### PR TITLE
implement greater than

### PR DESCRIPTION
**Pull request by supreme2580**
- [x] issue # https://github.com/keep-starknet-strange/shinigami/issues/14
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

Wrote code to check two integers for which is greater, if the first item is greater it returns 0, if the second item is greater it returns 1, wrote and made sure the test written for it pass.

### CODE REVIEW

File: src/opcodes/opcodes.cairo [hunk 1]
```diff
@@ -174,6 +188,16 @@ pub mod Opcode {
         engine.dstack.push_int(a + b);
     }
 
+    fn opcode_greaterthan(ref engine: Engine) {
+        let a = engine.dstack.pop_int();
+        let b = engine.dstack.pop_int();
+        engine.dstack.push_int(if a > b {
```
Comment (b-j-roberts) [inline src/opcodes/opcodes.cairo:194]: I believe this should returning 1 if b > a, how it is now if a == b you will get true. Might want to double check this

**Review by b-j-roberts — CHANGES_REQUESTED** [review]
Looks great! Can you add one more test for when comparing 2 of the same numbers?

File: src/opcodes/opcodes.cairo (hunk 1 above)
Comment (supreme2580) [reply src/opcodes/opcodes.cairo:194]: Okay I will work on this now

Comment (b-j-roberts) [reply src/opcodes/opcodes.cairo:194]: I think something like
```
if b > a {
    1
} else {
    0
}
```
Would be more efficient here.

Comment (supreme2580) [reply src/opcodes/opcodes.cairo:194]: Okay cool, will use that instead

**Pull request closed without merging**